### PR TITLE
#696: enable credential validation when using keyboard navigation

### LINKED ISSUE #696: BUG: Email/Phone accepts invalid email/phone
Issue: Email/Phone screen accepts invalid email/phone if user moves to next screen via the keyboard instead of the arrow buttons.

Task: The keyboard should require validation like the green arrow button does.

### PR DESCRIPTION
**Pull request by nationalkim**
closes #696 
criteria:
- keyboard "go" button requires same validation as the arrow button
- if credential is not valid, show error message

### CODE REVIEW

**Review by EngrTaofeek — CHANGES_REQUESTED** [review]
This looks great Natalie. The CustomToast was used in the old version of the app. I don't think we would want to use that in the 2.0 version. You can replace it with a snackbar or ignore the error message until the design team comes up with a custom error dialog.

File: app/src/main/java/org/greenstand/android/TreeTracker/signup/CredentialEntryView.kt [hunk 1]
```diff
@@ -109,7 +110,11 @@ fun CredentialEntryView(viewModel: SignupViewModel, state: SignUpState) {
                     ),
                     keyboardActions = KeyboardActions(
                         onGo = {
-                            viewModel.goToNameEntry()
+                            if (state.isEmailValid || state.isPhoneValid) {
+                                viewModel.goToNameEntry()
+                            } else {
+                                CustomToast.showToast("Please enter a valid email")
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/signup/CredentialEntryView.kt:116]: Please use string resources for any text.

**Review by Elforama — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/greenstand/android/TreeTracker/signup/CredentialEntryView.kt (hunk 1 above)
Comment (nationalkim) [reply app/src/main/java/org/greenstand/android/TreeTracker/signup/CredentialEntryView.kt:116]: @Elforama thanks! can you comment at all on which is the preferred way to go about things (i.e. making a snackbar or leaving a todo) regarding @EngrTaofeek 's suggestion above?

Comment (Elforama) [reply app/src/main/java/org/greenstand/android/TreeTracker/signup/CredentialEntryView.kt:116]: Let's use the snackbar.

**Comment by ZavenArra** [comment]
This PR has been incorporated elsewhere.  Thanks for the great work!

**Pull request closed without merging**